src/runtime-rs/crates/runtimes/virt_container/src/factory/mod.rs

```diff
@@ -0,0 +1,282 @@
+        template_path: toml_config.factory.template_path.clone(),
+        cache: toml_config.factory.vm_cache_number,
+        vm_cache: toml_config.factory.vm_cache_number > 0,
+        vm_config: VMConfig {
+            hypervisor_name: toml_config.runtime.hypervisor_name.clone(),
+            agent_name: toml_config.runtime.agent_name.clone(),
+            hypervisor_config: toml_config
+                .hypervisor
+                .get(&toml_config.runtime.hypervisor_name)
+                .cloned()
+                .unwrap_or_default(),
+            agent_config: toml_config
+                .agent
+                .get(&toml_config.runtime.agent_name)
+                .cloned()
+                .unwrap_or_default(),
+        },
+    };
+
+    // Template
+    if toml_config.factory.template {
+        new_factory(&mut factory_config, toml_config, true)
+            .await
+            .map_err(|e| {
+                error!(sl!(), "load vm factory failed: {:?}", e);
+                anyhow!(e).context("failed to load VM factory")
+            })?;
+
+        info!(sl!(), "begin destroy factory");
+
+        close_factory(&mut factory_config).map_err(|e| {
+            error!(sl!(), "Failed to close factory: {:?}", e);
+            anyhow!("Failed to close factory: {}", e)
+        })?;
+    } else {
+        info!(sl!(), "vm factory is not enabled");
+    }
+
+    info!(sl!(), "vm factory destroyed");
+    Ok(())
+}
+
+pub async fn status_factory_command() -> Result<()> {
+    println!("status_factory_command");
+
+    // Load Kata config
+    let (toml_config, _) = TomlConfig::load_from_default().context("load toml config")?;
+
+    // Build FactoryConfig
+    let mut factory_config = FactoryConfig {
+        template: toml_config.factory.template,
+        template_path: toml_config.factory.template_path.clone(),
+        cache: toml_config.factory.vm_cache_number,
+        vm_cache: toml_config.factory.vm_cache_number > 0,
+        vm_config: VMConfig {
+            hypervisor_name: toml_config.runtime.hypervisor_name.clone(),
+            agent_name: toml_config.runtime.agent_name.clone(),
+            hypervisor_config: toml_config
+                .hypervisor
+                .get(&toml_config.runtime.hypervisor_name)
+                .cloned()
+                .unwrap_or_default(),
+            agent_config: toml_config
+                .agent
+                .get(&toml_config.runtime.agent_name)
+                .cloned()
+                .unwrap_or_default(),
+        },
+    };
+
+    // Template
+    if toml_config.factory.template {
+        if new_factory(&mut factory_config, toml_config, true)
+            .await
+            .is_ok()
+        {
+            info!(sl!(), "vm factory is on");
+        } else {
+            info!(sl!(), "vm factory is off");
+        }
+    } else {
+        info!(sl!(), "vm factory is not enabled");
+    }
+
+    Ok(())
+}
+
+pub async fn new_factory(
```

Comment:
This is called by `init_factory_command`, `destroy_factory_command` and `status_factory_command`. However, these functions aren't referenced.